cpp/thirdparty/CMakeLists.txt

```diff
@@ -51,10 +52,82 @@ endif()
 
 # ## BOOST
 if(MEMILIO_USE_BUNDLED_BOOST)
-    include(BuildBoost)
+    # Ensure that the boost atomic library is used instead of the standard atomic library, where some functionality is only available as of C++20.
+    add_compile_definitions(BOOST_FILESYSTEM_NO_CXX20_ATOMIC_REF)
+    message(STATUS "Downloading Boost library. This may take some time.")
+
+    include(FetchContent)
+    FetchContent_Declare(boost
+    GIT_REPOSITORY https://github.com/boostorg/boost.git
+    GIT_TAG boost-${MEMILIO_BOOST_VERSION})
+    FetchContent_GetProperties(boost)
+
+    if(NOT boost_POPULATED)
+        FetchContent_Populate(boost)
+    endif()
+
+    if(NOT IS_DIRECTORY "${boost_SOURCE_DIR}/boost")
+        if(MSVC)
+            execute_process(COMMAND "bootstrap.bat"
+                WORKING_DIRECTORY ${boost_SOURCE_DIR}
+                )
+            # b2 produces a lot of uninformative output, suppress it.
+            execute_process(COMMAND "b2.exe" headers
+                WORKING_DIRECTORY ${boost_SOURCE_DIR}
+                OUTPUT_QUIET
+                ERROR_QUIET
+                )

```

Comment:
We talked about whether you need a developer command prompt to run bootstrap. Did you test this? 

Windows users expect to be able to configure the project e.g. from the cmake GUI just started from the start menu. If this doesn't work, it may be better to run bootstrap and b2 at build time, with `add_custom_command` or `add_custom_target`. 

There is also the option to load a cmake project directly into visual studio as well, at least one of these should work, prefereably both.

Reply:
still have to look for an alternative to the `OUTPUT_QUIET` option.